framework/src/markers/ReporterPointMarker.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReporterPointMarker.h"
+
+registerMooseObject("MooseApp", ReporterPointMarker);
+
+InputParameters
+ReporterPointMarker::validParams()
+{
+  InputParameters params = Marker::validParams();
+  params.addClassDescription("Marks the region inside or empty if it contains a reporter defined "
+                             "point for refinement or coarsening.");
+  params.addRequiredParam<ReporterName>("x_coord_name", "reporter x-coordinate name");
+  params.addRequiredParam<ReporterName>("y_coord_name", "reporter y-coordinate name");
+  params.addRequiredParam<ReporterName>("z_coord_name", "reporter z-coordinate name");
+  MooseEnum marker_states = Marker::markerStates();
+  params.addRequiredParam<MooseEnum>(
+      "inside", marker_states, "How to mark elements containing a point");
+  params.addRequiredParam<MooseEnum>(
+      "empty", marker_states, "How to mark elements not containing a point");
+  return params;
+}
+
+ReporterPointMarker::ReporterPointMarker(const InputParameters & parameters)
+  : Marker(parameters),
+    ReporterInterface(this),
+    _inside(parameters.get<MooseEnum>("inside").getEnum<MarkerValue>()),
+    _empty(parameters.get<MooseEnum>("empty").getEnum<MarkerValue>()),
+    _x_coord(getReporterValue<std::vector<Real>>("x_coord_name", REPORTER_MODE_REPLICATED)),
+    _y_coord(getReporterValue<std::vector<Real>>("y_coord_name", REPORTER_MODE_REPLICATED)),
+    _z_coord(getReporterValue<std::vector<Real>>("z_coord_name", REPORTER_MODE_REPLICATED))
+{
+}
+
+void
+ReporterPointMarker::timestepSetup()
+{
+  _pl = _fe_problem.mesh().getPointLocator();
+  _pl->enable_out_of_mesh_mode();
+  _point_elems.clear();
+  for (std::size_t i = 0; i < _x_coord.size(); ++i)
```

Comment:
You removed the check that ensures that the vectors all have the same size. This needs to happen in `timestepSetup`

Reply:
```suggestion
  const auto npoints = _x_coord.size();
  if (_y_coord.size() != npoints || _z_coord.size()_ != npoints)
    mooseError("Inconsistent coordinate component reporter vector sizes");
  for (std::size_t i = 0; i <npoints; ++i)
```

Reply:
why add 3 lines when you can add nine...

The temporary string is an anti-pattern, and IMO the size output is overkill. It should be enough to know that the sizes do not agree.

Reply:
I like the size because it tells what is actually wrong.  It will be easy to tell if one of the vectors is empty because it will give the size.  Otherwise it will be hard to know which vector is not being passed correctly.

Reply:
I disagree. This message is just flooding the user with data that is not particularly useful. Two things could have gone when this error is hit

1. The user is sourcing the coordinate columns from different reporters. That's 99.9% guaranteed a user mistake and knowing individual column sizes does not provide the user with any more useful information.
2. The reporter itself contains a logic error. This is serious and should be a rare event. This requires a developer to debug the issue, which won't be fine with this error message anyways.

Also, if the user really wants more details for whatever reason, they can just simply _output_ the reporter data.

Reply:
Not really resolved